main/git/APKBUILD

```diff
@@ -166,8 +166,16 @@ daemon() {
 		"$pkgdir"/$_gitcoredir/git-http-backend \
 		"$pkgdir"/$_gitcoredir/git-shell \
 		"$subpkgdir"/$_gitcoredir/
+}
+
+daemon_openrc() {
+	pkgdesc="Git protocol daemon (OpenRC init scripts)"
+	depends="git-daemon=${pkgver}-r${pkgrel} ${depends_openrc}"
```

Comment:
The depends on git-daemon does not make sense. This should be installed automatically after git-daemon.

Reply:
Well if as a user I install git-daemon-openrc; I would expect it to pull in git-daemon right? and openrc, as i want an openrc enabled git-daemon so i need both.
however, if I install git-daemon, I only want to pull in git-daemon-openrc if openrc is already part of my installation.
From other APKBUILDS I gather this to be what is needed?

Reply:
-openrc like -doc isn't typically meant to be installed by hand. So the direction of I'm installing -doc so install the origin package too isn't setup.

If you installed/install docs all the installed software triggers a bunch of -doc subpackages to be installed too. Similarly for the openrc package in this case.

Reply:
I get that, but as a user I have few options:
* install git-daemon (whilst openrc is already installed) -> i get git-daemon-openrc
* install git-daemon-openrc, which pulls in all dependencies (openrc, git-daemon)
* install git-daemon (whilst openrc is NOT installed) -> I get neither git-daemon-openrc nor openrc

I was under the impression that this solution would do exactly that; so where would this be going wrong then (ignoring the 'not using default_openrc case for the moment)?

Reply:
I believe you have those cases correct.

The second one (install git-daemon-openrc)  is not something that is typically handled as it is the opposite direction of how we expect these dependencies to work.

Reply:
well both scenario's should work now then :)

we do indeed and should expect what you suggest. A user will more then often have openrc installed and thus when installing the daemon will pull in the -openrc package.

In the container world; only the daemon bit will be installed, so without openrc no problem.

I think this is accomplished now, and we'd only go out of our way, to not support the use case where someone lazy just pulls daemon-openrc. It should 'just work' (tm) imo. Even if it is not expected.